src/V3Width.cpp

```diff
@@ -2150,27 +2150,32 @@ class WidthVisitor final : public VNVisitor {
         // Make sure dtype is sized
         nodep->dtypep(iterateEditMoveDTypep(nodep, nodep->subDTypep()));
         UASSERT_OBJ(nodep->dtypep(), nodep, "No dtype determined for var");
-        if (const AstUnsizedArrayDType* const unsizedp
-            = VN_CAST(nodep->dtypeSkipRefp(), UnsizedArrayDType)) {
-            if (!(m_ftaskp && m_ftaskp->dpiImport())) {
-                UINFO(9, "Unsized becomes dynamic array " << nodep << endl);
+        if (!(m_ftaskp && m_ftaskp->dpiImport())) {
+            AstNodeDType *dtp = nodep->dtypep();
+            AstDynArrayDType *np = nullptr;
+            while (const AstUnsizedArrayDType* const unsizedp
+                = VN_CAST(dtp->skipRefp(), UnsizedArrayDType)) {
+                UINFO(9, "Unsized becomes dynamic array ");
+                if (!np) {
+                    // for Var itself
+                    UINFONL(9, nodep << endl);
+                } else {
+                    // for subDType
+                    UINFONL(9, dtp << endl);
+                }
                 AstDynArrayDType* const newp
                     = new AstDynArrayDType{unsizedp->fileline(), unsizedp->subDTypep()};
-                AstDynArrayDType* np = newp;
-                AstNodeDType * cdtp = np->virtRefDTypep();
-                while (const AstUnsizedArrayDType* const cunsizedp
-                    = VN_CAST(cdtp->skipRefp(), UnsizedArrayDType)) {
-                    UINFO(9, "Unsized becomes dynamic array " << cdtp << endl);
-                    AstDynArrayDType* const cnewp
-                        = new AstDynArrayDType{cunsizedp->fileline(), cunsizedp->subDTypep()};
-                    np->refDTypep(cnewp);
-                    v3Global.rootp()->typeTablep()->addTypesp(cnewp);
-                    np = cnewp;
-                    cdtp = np->virtRefDTypep();
+                newp->dtypep(newp);
```

Comment:
This line instead of redoing `nodep->dtypep(iterateEditMoveDTypep(nodep, nodep->subDTypep()))`